src/GafferScene/LightToCamera.cpp

```diff
@@ -0,0 +1,312 @@
+	addChild( new IntPlug( "resolution", Plug::In, 512 ) );
+
+	// Fast pass-throughs for things we don't modify
+	outPlug()->boundPlug()->setInput( inPlug()->boundPlug() );
+}
+
+LightToCamera::~LightToCamera()
+{
+}
+
+Gaffer::IntPlug *LightToCamera::resolutionPlug()
+{
+	return getChild<IntPlug>( g_firstPlugIndex );
+}
+
+const Gaffer::IntPlug *LightToCamera::resolutionPlug() const
+{
+	return getChild<IntPlug>( g_firstPlugIndex );
+}
+
+void LightToCamera::affects( const Gaffer::Plug *input, AffectedPlugsContainer &outputs ) const
+{
+	SceneElementProcessor::affects( input, outputs );
+
+	if( input == resolutionPlug() )
+	{
+		outputs.push_back( outPlug()->objectPlug() );
+	}
+	else if( input == inPlug()->attributesPlug() )
+	{
+		outputs.push_back( outPlug()->objectPlug() );
+	}
+	else if( input == outPlug()->objectPlug() )
+    {
+        outputs.push_back( outPlug()->transformPlug() );
+    }
+}
+
+bool LightToCamera::processesObject() const
+{
+	return true;
+}
+
+void LightToCamera::hashProcessedObject( const ScenePath &path, const Gaffer::Context *context, IECore::MurmurHash &h ) const
+{
+	resolutionPlug()->hash( h );
+	inPlug()->attributesPlug()->hash( h );
+}
+
+IECore::ConstObjectPtr LightToCamera::computeProcessedObject( const ScenePath &path, const Gaffer::Context *context, IECore::ConstObjectPtr inputObject ) const
+{
+	IECore::ConstCameraPtr camera = convertLightToCamera( inPlug()->attributesPlug()->getValue(), resolutionPlug()->getValue() );
+
+	if( camera )
+	{
+		return camera;
+	}
+
+	return inputObject;
+}
+
+bool LightToCamera::processesTransform() const
+{
+	return true;
+}
+
+void LightToCamera::hashProcessedTransform( const ScenePath &path, const Gaffer::Context *context, IECore::MurmurHash &h ) const
+{
+	hashProcessedObject( path, context, h );
+}
+
+M44f LightToCamera::computeProcessedTransform( const ScenePath &path, const Gaffer::Context *context, const M44f &inputTransform ) const
+{
+	ConstObjectPtr object = outPlug()->objectPlug()->getValue();
+	if( const Camera *camera = runTimeCast<const Camera>( object.get() ) )
+	{
+		CompoundDataMap::const_iterator it = camera->parameters().find( "__projectionTransformOffset" );
+		if( it != camera->parameters().end() )
+		{
+			if( const M44fData *transformOffset = runTimeCast<const M44fData>( it->second.get() ) )
+			{
+				return transformOffset->readable() * inputTransform;
+			}
+		}
+	}
+	return inputTransform;
+}
+
+bool LightToCamera::processesAttributes() const
+{
+	return true;
+}
+
+void LightToCamera::hashProcessedAttributes( const ScenePath &path, const Gaffer::Context *context, IECore::MurmurHash &h ) const
+{
+	// Attributes depend only on input attributes
+}
+	
+
+IECore::ConstCompoundObjectPtr LightToCamera::computeProcessedAttributes( const ScenePath &path, const Gaffer::Context *context, IECore::ConstCompoundObjectPtr inputAttributes ) const
+{
+	// Remove any light attributes - because we may be shifting the transform, they could be invalidated,
```

Comment:
I don't think this needs to be the justication - we're converting lights to cameras, so the lights should be deleted regardless of any shifting transforms.